x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportPutTrainedModelAction.java

```diff
@@ -138,11 +141,16 @@ protected void masterOperation(Task task,
                     minCompatibilityVersion.toString()));
                 return;
             }
+        } else if (state.nodes().getMinNodeVersion().before(state.nodes().getMaxNodeVersion())
```

Comment:
Could we move this check at the top of `masterOperation`?

In addition, do we really need this check? I'm trying to think what happens if a user starts a rolling upgrade to the cluster and installs a fleet package that tries to put a model with `defer_definition_compression`. Is it worth failing the request? If we allowed it what would break? I assume if the model was loaded in an older node it would fail as the definition would be missing. Is that preferable?

Reply:
@dimitris-athanasiou I am being paranoid for sure. My concern is that we have no way of determining if the model definition can be inflated on the current min node version or not. Previously, we were able to validate that. I guess its "buyer beware" and I can remove this check, but they could get an ugly parsing error. Which, I suppose, is the case anyways.